core/lib/zksync_core/src/api_server/tx_sender/proxy.rs

```diff
@@ -65,10 +67,28 @@ impl TxCache {
     async fn run_updates(
         self,
         pool: ConnectionPool<Core>,
-        stop_receiver: watch::Receiver<bool>,
+        mut stop_receiver: watch::Receiver<bool>,
     ) -> anyhow::Result<()> {
         const UPDATE_INTERVAL: Duration = Duration::from_secs(1);
 
+        tracing::info!(
+            "Waiting for at least one L1 batch in Postgres to start TxCache::run_updates"
+        );
+        // Starting the server before L1 batches are present in Postgres can lead to some invariants the server logic
```

Comment:
Nit: It's not the "server" being started here, but rather transaction cache updater.